WalletWasabi.Fluent/Controls/Dialog.axaml.cs

```diff
@@ -197,20 +200,60 @@ private bool ShowAlert
 		set => SetValue(ShowAlertProperty, value);
 	}
 
-	private CancellationTokenSource? CancelPointerPressedDelay { get; set; }
+	private CancellationTokenSource? CancelPointerOpenedPressedDelay { get; set; }
 
-	private void UpdateDelay(bool isDialogOpen)
+	private CancellationTokenSource? CancelPointerActivatedPressedDelay { get; set; }
+
+	protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
+	{
+		base.OnAttachedToVisualTree(e);
+
+		_updateActivatedDelayDisposable = ApplicationHelper.MainWindowActivated.Subscribe(UpdateActivatedDelay);
+	}
+
+	protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e)
+	{
+		base.OnDetachedFromVisualTree(e);
+
+		_updateActivatedDelayDisposable?.Dispose();
+	}
+
+	private void UpdateOpenedDelay(bool isDialogOpen)
 	{
 		try
 		{
-			_canCancelOnPointerPressed = false;
-			CancelPointerPressedDelay?.Cancel();
+			_canCancelOpenedOnPointerPressed = false;
+			CancelPointerOpenedPressedDelay?.Cancel();
 
 			if (isDialogOpen)
 			{
-				CancelPointerPressedDelay = new CancellationTokenSource();
+				CancelPointerOpenedPressedDelay = new CancellationTokenSource();
+
+				Task.Delay(TimeSpan.FromSeconds(1), CancelPointerOpenedPressedDelay.Token).ContinueWith(_ => _canCancelOpenedOnPointerPressed = true);
+			}
+		}
+		catch (OperationCanceledException)
+		{
+			// ignored
+		}
+	}
+
+	private void UpdateActivatedDelay(bool isWindowActivated)
+	{
+		try
+		{
+			if (!isWindowActivated)
+			{
+				_canCancelActivatedOnPointerPressed = false;
+			}
+
+			CancelPointerActivatedPressedDelay?.Cancel();
+
+			if (isWindowActivated)
+			{
+				CancelPointerActivatedPressedDelay = new CancellationTokenSource();
 
-				Task.Delay(TimeSpan.FromSeconds(1), CancelPointerPressedDelay.Token).ContinueWith(_ => _canCancelOnPointerPressed = true);
+				Task.Delay(TimeSpan.FromSeconds(5), CancelPointerActivatedPressedDelay.Token).ContinueWith(_ => _canCancelActivatedOnPointerPressed = true);
```

Comment:
5 seconds is too long, we only need to prevent the cancellation at that moment. 1 second should be enough.

Reply:
1 second seemed to short when user switches from other window and after window is activated, maybe 2 seconds at least?

Reply:
We want to prevent the click when the user's intention was to give back the focus to the software. Actually, an even shorter period should be enough. I would go with 1 sec just for consistency.

Or am I missing something? Why is it better to prevent it for a longer time?

Reply:
Well I was testing and it takes a bit of time until you switch back to window after it's activated so I could still accidently click and dismiss dialog.

Reply:
> it takes a bit of time until you switch back to window after it's activated

How did you test it? I don't understand this part.

The original issue is talking about when you want to switch back to the wasabi window with a click. This is what we want to prevent:
![gif1](https://user-images.githubusercontent.com/16364053/224974397-45e974d8-6904-4b2d-a763-90fb3bfbd7d5.gif)

In this case, the user miss-clicked, they just wanted to give Wasabi the focus. We don't have to prevent it for a longer time.

Reply:
I was switching to different window that was above wasabi or second case minimizing wasabi window and going to another application so it takes more time to get back to main wasabi window after it was activated again. If we want only solve the above issue 1 second delay is enough.

Reply:
I have reduced delay to 1 second.